plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx

```diff
@@ -0,0 +1,164 @@
+    | {
+        kind?: string;
+        namespace?: string;
+        name: string;
+      },
+): string {
+  // Simplify the diagram output by hiding only the default namespace
+  return serializeEntityRef(ref)
+    .toString()
+    .toLowerCase()
+    .replace(':default/', ':');
+}
+
+/**
+ * Dynamically generates a diagram of a system, its assigned entities,
+ * and relationships of those entities.
+ */
+export function SystemDiagramCard({ entity }: SystemDiagramProps) {
+  const currentSystemName = entity.metadata.name;
+  const currentSystemNode = simplifiedEntityName(entity);
+  const systemNodes = new Array<{ id: string }>();
+  const systemEdges = new Array<{ from: string; to: string; label: string }>();
+
+  const catalogApi = useApi(catalogApiRef);
+  const { loading, error, value: catalogResponse } = useAsync(() => {
+    return catalogApi.getEntities({
+      filter: {
+        kind: ['Component', 'API', 'Resource', 'System', 'Domain'],
+        ['spec.system']: currentSystemName,
+      },
+    });
+  }, [catalogApi]);
+
+  // pick out the system itself
+  systemNodes.push({
+    id: currentSystemNode,
+  });
+
+  // check if the system has an assigned domain
+  // even if the domain object doesn't exist in the catalog, display it in the map
+  const catalogItemDomain = getEntityRelations(entity, RELATION_PART_OF, {
+    kind: 'domain',
+  });
+  catalogItemDomain.forEach(foundDomain =>
+    systemNodes.push({
+      id: simplifiedEntityName(foundDomain),
+    }),
+  );
+  catalogItemDomain.forEach(foundDomain =>
+    systemEdges.push({
+      from: currentSystemNode,
+      to: simplifiedEntityName(foundDomain),
+      label: 'part of',
+    }),
+  );
+
+  if (catalogResponse && catalogResponse.items) {
+    for (const catalogItem of catalogResponse.items) {
+      systemNodes.push({
+        id: simplifiedEntityName(catalogItem),
+      });
+
+      // Check relations of the entity assigned to this system to see
+      // if it relates to other entities.
+      // Note those relations may, or may not, be explicitly
+      // assigned to the system.
+      const catalogItemRelations_partOf = getEntityRelations(
+        catalogItem,
+        RELATION_PART_OF,
+      );
+      catalogItemRelations_partOf.forEach(foundRelation =>
+        systemEdges.push({
+          from: simplifiedEntityName(catalogItem),
+          to: simplifiedEntityName(foundRelation),
+          label: 'part of',
+        }),
+      );
+
+      const catalogItemRelations_providesApi = getEntityRelations(
+        catalogItem,
+        RELATION_PROVIDES_API,
+      );
+
+      catalogItemRelations_providesApi.forEach(foundRelation =>
+        systemEdges.push({
+          from: simplifiedEntityName(catalogItem),
+          to: simplifiedEntityName(foundRelation),
+          label: 'provides API',
+        }),
+      );
+    }
+  }
+
+  if (loading) {
+    return <Progress />;
+  } else if (error) {
+    return <Alert severity="error">{error.message}</Alert>;
```

Comment:
Could through in the new error panel here, but don't wanna block on that

Reply:
![image](https://user-images.githubusercontent.com/33203301/112484004-ee9afe80-8d4f-11eb-9c27-e746d7d5199c.png)

Reply:
Interestingly, may have found a bug with the new warningpanel changes that separate out the error object. It seems to occasionally overwrite the supplied title but not always. Will dig into later.